subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -337,6 +337,26 @@
      */
     Dependency localGroovy();
 
+    /**
+     * Returns the dependency constraint handler for this project.
+     *
+     * @return the dependency constraint handler for this project
+     * @since 4.5
+     */
+    @Incubating
+    DependencyConstraintHandler getConstraints();
+
+    /**
+     * Configures dependency constraint for this project.
+     *
+     * <p>This method executes the given action against the {@link org.gradle.api.artifacts.dsl.DependencyConstraintHandler} for this project.</p>
+     *
+     * @param configureAction the action to use to configure module metadata
+     * @since 4.5
+     */
+    @Incubating
+    void constraints(Action<? super DependencyConstraintHandler> configureAction);
+
```

Comment:
@bamboo Do we need to do anything for `DependencyHandlerScope` in Kotlin DSL to reflect these interface additions? Or do we just need to build a new Kotlin DSL version against the extended interface?